spec/unit/puppet/provider/dsc_base_provider/dsc_base_provider_spec.rb

```diff
@@ -590,6 +590,49 @@
       end
     end
 
+    context 'when handling DateTimes' do
+      before(:each) do
+        allow(ps_manager).to receive(:execute).with(script).and_return({ stdout: 'DSC Data' })
+        allow(JSON).to receive(:parse).with('DSC Data').and_return(parsed_invocation_data)
+        allow(provider).to receive(:fetch_cached_hashes).and_return([])
+      end
+
+      context 'When the DateTime is nil' do
+        let(:name_hash) { { name: 'foo', dsc_name: 'foo', dsc_time: nil } }
+        let(:parsed_invocation_data) do
+          { 'Name' => 'foo', 'Ensure' => 'Present', 'Time' => nil }
+        end
+
+        it 'returns nil for the value' do
+          expect(result[:dsc_time]).to eq(nil)
+        end
+      end
+
+      context 'When the DateTime is an invalid string' do
+        let(:name_hash) { { name: 'foo', dsc_name: 'foo', dsc_time: '21000101' } }
```

Comment:
Let's make it more obvious this is a bad string and make it `foo` or something

Reply:
Done